src/adapters/mc/complianceengine/Baseline.c

```diff
@@ -3,10 +3,36 @@
 
 #include "../Common.h"
 #include "ComplianceEngineInterface.h"
+#include <assert.h>
 
 static MMI_HANDLE gComplianceEngine = NULL;
 
-static const char gComponentName[] = "ComplianceEngine";
+// This function is used to defer the call to ComplianceEngineMmiOpen
+// past the call to BaselineInitialize which is the library constructor.
+// The problem with initialization is we may use a statically initialized
+// data which requires C++ constructors to be called (e.g. static const std::map<...>).
+// If we called the ComplianceEngineMmiOpen inside the BaselineInitialize
+// function body, we could have experienced unexpected crashes due to
+// initialization order fiasco.
+static void InitializeHandle()
```

Comment:
Could we avoid this problem altogether by replacing global variables with function local statics:
```
const std::map& GetXXX()
{
    static std::map<....> smthMap;
    return smthMap;
}
```